### PR TITLE
Documentation for SelfTargetExecutor

### PR DESCRIPTION
**Pull request by AlfinST**
# Summary

@yahoo/ychaos-dev

Adding SelfTargetExecutor docs 
---

**Fixes**: #12 

#### Prerequisites
- [x] I have read the contribution guidelines

#### Code Analysis
- [ ] Covered by Unittests
- [ ] Security warnings ignored (Why?!)

#### Project related
- [ ] Schema changed and is backward compatible.
- [ ] Introduces a new sub-command for ychaos CLI

### Autogenerated Files
- [ ] Auto generated schema

### Pre-Merge Checklist

- [ ] All the build jobs are passing

---

### Checklist (Reviewer 1)

- [ ] Reviewed Source Code
- [ ] Reviewed Tests (If applicable)
- [ ] Reviewed Documentation (If applicable)

### Checklist (Reviewer 2)

- [ ] Reviewed Source Code
- [ ] Reviewed Tests (If applicable)
- [ ] Reviewed Documentation (If applicable)

### CODE REVIEW

**Comment by shashankrnr32** [comment]
@AlfinST Please add example usage as well to the documentation

File: docs/testplan/target_spec.md [hunk 1]
```diff
@@ -24,4 +24,7 @@ each of the target types available in the package.
         heading_level: 4
         attributes: False
         show_root_full_path: False
-   
\ No newline at end of file
+   
+## Localhost (`self`)
```
Comment (shashankrnr32) [inline docs/testplan/target_spec.md:28]:
```suggestion
## Self (`self`)
```

Lets not use the term "Localhost"

File: docs/testplan/target_spec.md [hunk 2]
```diff
@@ -24,4 +24,7 @@ each of the target types available in the package.
...
+
+::: ychaos.testplan.attack.SelfTargetDefinition
```
Comment (shashankrnr32) [inline docs/testplan/target_spec.md:30]: Add the `mkdocstrings` Filters like the above machine configurations

```
selection:
        filters:
        - "!"
    rendering:
        heading_level: 4
        attributes: False
        show_root_full_path: False
```

File: docs/testplan/sample_test_plan.md [hunk 3]
```diff
@@ -19,6 +19,16 @@ attack:
         duration: 60 # Burn CPU for 60s
 ```
 
+```yaml
+description: Increase CPU utilisation on all the cores for 60 seconds on local machine
+attack:
+  target_type: self
+  agents:
```
Comment (shashankrnr32) [inline docs/testplan/sample_test_plan.md:26]:
```suggestion
  target_type: self
  # target_config not needed for this target_type
  agents:
```

**Review by shashankrnr32 — COMMENTED** [review]
Very minor comments. Looks good

**Review by shashankrnr32 — APPROVED** [review]
LGTM

**Review by vijaybabu4589 — APPROVED** [review]

**Pull request merged**